navigation_and_routing/lib/src/screens/sign_in.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+
+class Credentials {
+  final String username;
+  final String password;
+  Credentials(this.username, this.password);
+}
+
+class SignInScreen extends StatefulWidget {
+  final ValueChanged<Credentials> onSignIn;
+
+  const SignInScreen({
+    required this.onSignIn,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  _SignInScreenState createState() => _SignInScreenState();
+}
+
+class _SignInScreenState extends State<SignInScreen> {
+  String username = '';
+  String password = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Center(
+        child: Card(
+          child: Container(
+            constraints: BoxConstraints.loose(const Size(600, 600)),
+            padding: const EdgeInsets.all(8),
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              mainAxisSize: MainAxisSize.min,
+              children: [
+                Text('Sign in', style: Theme.of(context).textTheme.headline4),
+                TextField(
+                  decoration: const InputDecoration(labelText: 'Username'),
+                  onChanged: (v) {
+                    setState(() {
```

Comment:
Instead of rebuilding this widget on every keypress, maybe use two TextEditingController and grab their value only when the user hits sign-in?